File: src/python/_0605_Can_Place_Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        i = 1
        flowerbed = [0] + flowerbed + [0]

        while i < len(flowerbed) - 1:
            if flowerbed[i - 1] + flowerbed[i] + flowerbed[i + 1] == 0:
                n -= 1
                i += 1
            i += 1

        return n <= 0

File: src/java/_0605_Can_Place_Flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int length = flowerbed.length;

        for (int i = 0; i < length; i++) {
            if (flowerbed[i] == 0) {
                int prev = (i == 0) ? 0 : flowerbed[i - 1];
                int next = (i == length - 1) ? 0 : flowerbed[i + 1];
                if (prev == 0 && next == 0) {
                    n--;
                    flowerbed[i] = 1;
                }
            }
        }

        return n <= 0;
    }
}
